### PR TITLE
[WIP] External IP Support

### PR DESCRIPTION
**Pull request by mcastelino**
External IP support in the CNCI agent. 

The CNCI agent is now able to support setting up of External IP (also called Public IPs) to Tenant Local Instance IPs. The scheduler send the CNCI Public IP commands onwards to the CNCI specified by the controller. Controller and CLI are enhanced to manage the external IP pools and send the commands to the CNCI via Scheduler to perform the operations.

### CODE REVIEW

File: networking/ciao-cnci-agent/network.go [hunk 1]
```diff
@@ -306,13 +306,13 @@ func unmarshallPubIP(cmd *payloads.PublicIPCommand) (net.IP, net.IP, error) {
...
 	if err != nil {
-		glog.Errorf("cnci.assignPubIP invalid params %v %v", err, cmd)
+		return fmt.Errorf("cnci.assignPubIP invalid params %v %v", err, cmd)
 	}
 
-	if enableNetwork {
-		glog.Infof("cnci.assignPubIP success %v %v %v", prIP, puIP, cmd)
+	err = gFw.PublicIPAccess(libsnnet.FwEnable, prIP, puIP, gCnci.ComputeLink[0].Attrs().Name)
+	if err != nil {
+		return fmt.Errorf("%v", err)
```
Comment (markdryan) [inline networking/ciao-cnci-agent/network.go]: Why don't we just return err?

**Comment by markdryan** [comment]
@mcastelino Looks okay to me although I must admit I don't understand too much of the code.  We still need @tpepper's approval for the scheduler changes I guess.

File: networking/ciao-cnci-agent/network.go (hunk 1 above)
Comment (mcastelino) [inline networking/ciao-cnci-agent/network.go]: @markdryan missed this when I cleaned up the patches. Will add one extra patch on top of leo's work to fix this and other issues... I wanted to keep most of his work intact and just combined some his patches into logical sets.

**Comment by mcastelino** [comment]
@tpepper can you review the scheduler changes please. I merged Leo's work with yours. Hopefully I got it right

**Comment by mcastelino** [comment]
@tpepper @markdryan Unaddressed review comments will be addressed as part of fixing https://github.com/01org/ciao/issues/879

**Pull request closed without merging**